src/contracts/atlas/AtlasVerification.sol

```diff
@@ -191,31 +135,17 @@ contract AtlasVerification is EIP712, DAppIntegration {
 
                 SolverOperation memory solverOp = solverOps[i];
 
-                if (tx.gasprice > solverOp.maxFeePerGas) {
-                    // Verify solver max gas price
-                    continue;
-                }
+                if (tx.gasprice > solverOp.maxFeePerGas) continue;
 
-                if (block.number > solverOp.deadline) {
-                    // Verify solver deadline
-                    continue;
-                }
+                if (block.number > solverOp.deadline) continue;
 
-                if (solverOp.from == userOp.from) {
-                    continue;
-                }
+                if (solverOp.from == userOp.from) continue;
 
-                if (solverOp.to != ATLAS) {
-                    continue;
-                }
+                if (solverOp.to != ATLAS) continue;
 
-                if (solverOp.solver == ATLAS || solverOp.solver == address(this)) {
-                    continue;
-                }
+                if (solverOp.solver == ATLAS || solverOp.solver == address(this)) continue;
 
-                if (solverOp.userOpHash != userOpHash) {
-                    continue;
-                }
+                if (solverOp.userOpHash != userOpHash) continue;
 
                 // If all initial checks succeed, add solver op to new array
                 prunedSolverOps[i] = solverOp;
```

Comment:
```suggestion
                prunedSolverOps[validSolverCount] = solverOp;
```
If we rather use `validSolverCount` as the index to store the valid solver ops, it should pack all the valid ones together from the left of the array, as opposed to possibly having empty element gaps between valid solver ops scattered throughout the array, as it would with `i`.

This could reduce gas costs elsewhere in Atlas because then we know we can stop iterating through the array when we hit an empty element.

Reply:
To complete the optimization, I think it would just be:
`if (solverOps[winningSearcherIndex].from == address(0)) break;`
and remove the other solverOps.from check, in the loop in `_execute()`

Reply:
The one issue with this approach is that we'd be changing the indexes of the solver operations, which may impact the simulator and the settlement system (ie we track winning searcher index)

I think some of those concerns no longer apply though - if there's no issues with changing the index then I'm all for it.

Reply:
We calculate the winningSearcherIndex using the modified solverOps array, so it would already be in the "packed from the left" form and the winning index would map to that form. So from what I can see settlement should be okay.

`AtlasVerification.validCalls()` returns the original solverOps array before this left-packing, if `isSimulation == true`. So that might be a problem. Could we just move that isSimulation check and return to after the solverOps sorting? 

I'll make a branch from this PR to show the specific changes for this idea and we can decide from that if its safe

Reply:
Okay, the attempt to safely implement the optimization is here: https://github.com/FastLane-Labs/atlas/pull/74